### PR TITLE
 add --conf parameter

### PR DESCRIPTION
**Pull request by arowser**
#1526 Custom path for config file

### CODE REVIEW

File: lightningd/options.c [hunk 1]
```diff
@@ -697,7 +700,12 @@ static void opt_parse_from_config(struct lightningd *ld)
 	char *argv[3];
 	int i, argc;
 
-	contents = grab_file(ld, "config");
+	if (ld->config_filename!= NULL) {
+		if (!path_is_file(ld->config_filename))
+			fatal("Configuration file: %s not exists.", ld->config_filename);
```
Comment (rustyrussell) [inline lightningd/options.c]: This test is unnecessary?

File: lightningd/options.c [hunk 2]
```diff
@@ -697,7 +700,12 @@ static void opt_parse_from_config(struct lightningd *ld)
...
+		contents = grab_file(ld, ld->config_filename);
```
Comment (rustyrussell) [inline lightningd/options.c]: Do this by itself, then below...

File: lightningd/options.c [hunk 3]
```diff
@@ -697,7 +700,12 @@ static void opt_parse_from_config(struct lightningd *ld)
...
+	} else
+		contents = grab_file(ld, "config");
 	/* Doesn't have to exist. */
 	if (!contents) {
 		if (errno != ENOENT)
```
Comment (rustyrussell) [inline lightningd/options.c]: `if (errno != ENOENT || ld->config_filename != NULL)`

File: lightningd/lightningd.h [hunk 4]
```diff
@@ -80,6 +80,8 @@ struct lightningd {
 	char *config_dir;
 	char *rpc_filename;
 
+	/* Configuration file name */
+	char *config_filename;
```
Comment (rustyrussell) [inline lightningd/lightningd.h]: You need to initialize this to NULL in lightningd.c new_lightningd(), too.

Comment (arowser) [reply lightningd/lightningd.h]: Thanks, fixed all

File: lightningd/options.c [hunk 5]
```diff
@@ -697,10 +700,13 @@ static void opt_parse_from_config(struct lightningd *ld)
 	char *argv[3];
 	int i, argc;
 
-	contents = grab_file(ld, "config");
+	if (ld->config_filename!= NULL) {
```
Comment (cdecker) [inline lightningd/options.c]: Nit: whitespace

File: lightningd/options.c [hunk 6]
```diff
@@ -697,10 +700,13 @@ static void opt_parse_from_config(struct lightningd *ld)
...
+		contents = grab_file(ld, ld->config_filename);
+	} else
+		contents = grab_file(ld, "config");
 	/* Doesn't have to exist. */
 	if (!contents) {
-		if (errno != ENOENT)
+		if ((errno != ENOENT) || ld->config_filename != NULL)
```
Comment (cdecker) [inline lightningd/options.c]: A comment here would be much appreciated: in the default config case we allow it not to exist, but if the config was specified on the command line it has to exist.

Comment (cdecker) [reply lightningd/options.c]: Also inconsistent parentheses.

**Comment by cdecker** [comment]
Squashing this into one commit would be nice :wink:

**Comment by cdecker** [comment]
ACK 57e2c6d

**Pull request merged**